File: leetcode/7.整数反转.java
/*
 * @lc app=leetcode.cn id=7 lang=java
 *
 * [7] 整数反转
 */

// @lc code=start
class Solution {
    public int reverse(int x) {
        int res = 0;
        while (x != 0) {
            if (res < Integer.MIN_VALUE / 10 || res > Integer.MAX_VALUE / 10)
                return 0;
            int temp = x % 10;
            x = x / 10;
            res = 10 * res + temp;
        }
        return res;
    }

    // public static void main(String[] args) {
    //     Solution s = new Solution();
    //     System.out.println(s.reverse(1234));
    // }
}
// @lc code=end

File: leetcode/7.整数反转.go
package main

import (
	"fmt"
	"math"
)

/*
 * @lc app=leetcode.cn id=7 lang=golang
 *
 * [7] 整数反转
 */

// @lc code=start
func reverse(x int) int {
	res := 0
	for x != 0 {
		if res < math.MinInt32/10 || res > math.MaxInt32/10 {
			return 0
		}
		temp := x % 10
		x = x / 10
		res = res*10 + temp
	}
	return res
}

func main() {
	fmt.Println(reverse(1234))
}

// @lc code=end
